### PR TITLE
fix(starlette): Stop duplicating `scope["root_path"]` in URLs

### LINKED ISSUE #6577: ASGI integration doubles the mount prefix in request.url for Starlette Mounted sub-apps (root_path + path)
### Summary

`_get_url()` in `sentry_sdk/integrations/_asgi_common.py` builds the request URL as `root_path + path`:

```python
path = asgi_scope.get("root_path", "") + asgi_scope.get("path", "")
```

Under the current ASGI spec, `scope["path"]` **already includes** `scope["root_path"]`. So for any Starlette `Mount(...)`ed sub-app the mount prefix is counted twice — a request to `/mcp/` is reported as `/mcp/mcp/`. This affects `request.url` on both error and transaction events for every sub-mounted ASGI app.

### Version / environment

- sentry-sdk: `2.59.0` (also reproduces on `master` — `_get_url` is unchanged there)
- starlette: `1.0.1`
- uvicorn: `0.38.0`
- python: `3.14.5`

### Minimal reproduction

```python
from sentry_sdk.integrations._asgi_common import _get_url

# The scope a Starlette Mount("/mcp", sub_app) hands its sub-app when a client
# requests "/mcp/" — path includes root_path, per the current ASGI spec.
scope = {
    "type": "http", "scheme": "http",
    "server": ("api.example.com", 80),
    "headers": [(b"host", b"api.example.com")],
    "root_path": "/mcp",
    "path": "/mcp/",
}
print(_get_url(scope, "http", "api.example.com"))
# ->  http://api.example.com/mcp/mcp/        (expected: http://api.example.com/mcp/)
```

Confirming a real `Mount` produces exactly that scope:

```python
from starlette.applications import Starlette
from starlette.routing import Mount, Route
from starlette.responses import PlainTextResponse
from starlette.testclient import TestClient

seen = {}
async def ep(request):
    seen.update(path=request.scope["path"], root_path=request.scope["root_path"])
    return PlainTextResponse("ok")

app = Starlette(routes=[Mount("/mcp", app=Starlette(routes=[Route("/", endpoint=ep)]))])
TestClient(app).get("/mcp/")
print(seen)   # {'path': '/mcp/', 'root_path': '/mcp'}
```

### Root cause

The ASGI spec states that `scope["path"]` contains the full path *including* `root_path` (see [starlette#1336](https://github.com/encode/starlette/issues/1336) and [uvicorn#2213](https://github.com/encode/uvicorn/pull/2213)). `_get_url` predates that change and assumes the WSGI-style disjoint `SCRIPT_NAME` + `PATH_INFO` split, so it double-counts whenever `root_path` is non-empty — which Starlette sets for every `Mount`.

### Relationship to prior reports

- #1832 reported the same `root_path + path` doubling but was closed inconclusively because it couldn't be reproduced — that repro used FastAPI's `root_path=` argument, where (pre-uvicorn#2213) the prefix was *not* folded into `path`, so the sum happened to be correct. The trigger that actually doubles is a **Starlette `Mount`**, shown above.
- #2631 (mounted sub-apps) was closed as completed, but `request.url` doubling persists — `_get_url` still sums the two.

### Suggested fix

Only prepend `root_path` when `path` doesn't already include it:

```python
path = asgi_scope.get("path", "")
root_path = asgi_scope.get("root_path", "")
if root_path and not (path == root_path or path.startswith(root_path + "/")):
    path = root_path + path
```

This preserves the old behavior for servers that still pass a disjoint `root_path` / `path`, while not double-counting for spec-compliant ones.

### PR DESCRIPTION
**Pull request by alexander-alderman-webb**
### Description

Add the `root_path_in_path` parameter to `_get_url()` and related functions.

If `root_path_in_path` is `_RootPathInPath.EXCLUDED`, the existing behavior that prepends `scope["root_path"]` when forming the url is preserved.

The other `_RootPathInPath.EITHER` option is used by the Starlette integration. It preserves Starlette’s behavior of accepting ASGI scopes where `scope["path"]` either includes `scope["root_path"]` or is already relative to it, using heuristics that match the libraries logic.

#### Issues
Closes https://github.com/getsentry/sentry-python/issues/6577

#### Reminders
- Please add tests to validate your changes, and lint your code using `uv run ruff`.
- Add GH Issue ID _&_ Linear ID (if applicable)
- PR title should use [conventional commit](https://develop.sentry.dev/engineering-practices/commit-messages/#type) style (`feat:`, `fix:`, `ref:`, `meta:`)
- For external contributors: [CONTRIBUTING.md](https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md), [Sentry SDK development docs](https://develop.sentry.dev/sdk/), [Discord community](https://discord.gg/Ww9hbqr)

### CODE REVIEW

File: sentry_sdk/integrations/asgi.py [hunk 1]
```diff
@@ -116,6 +117,7 @@ def __init__(
         span_origin: str = "manual",
         http_methods_to_capture: "Tuple[str, ...]" = DEFAULT_HTTP_METHODS_TO_CAPTURE,
         asgi_version: "Optional[int]" = None,
+        path_includes_root_path: bool = True,
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/asgi.py]: good point, will fix

File: sentry_sdk/integrations/quart.py [hunk 2]
```diff
@@ -86,16 +87,25 @@ def setup_once() -> None:
...
+    version = package_version("quart")
+
     async def sentry_patched_asgi_app(
         self: "Any", scope: "Any", receive: "Any", send: "Any"
     ) -> "Any":
-        if sentry_sdk.get_client().get_integration(QuartIntegration) is None:
+        if (
+            sentry_sdk.get_client().get_integration(QuartIntegration) is None
+            or version is None
+        ):
```
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/quart.py]: no data is better than wrong data

[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/quart.py]: no data is better than wrong data

**Review by wahajahmed010 — COMMENTED** [review]
## Review: getsentry/sentry-python#6579

**Summary:** Clean fix for ASGI URL duplication when `scope["root_path"]` is already included in `scope["path"]`. The approach is sound — thread a `path_includes_root_path` boolean through the URL construction pipeline.

### ✅ What works well
- **Backward compatible** — default is `True`, preserving existing behavior for frameworks that don't include root_path in path
- **Framework-specific handling** — correctly version-gated for Starlette (>=0.33), Quart (>=0.19), and hardcoded for Litestar/Starlite
- **Test coverage** — all 6 integrations tested (Django ASGI, FastAPI, Litestar, Quart, Starlette, Starlite), both span streaming and static lifecycle modes
- **Clean threading** — parameter flows through `_get_url` → `_get_request_data` → `_get_request_attributes` → `_get_transaction_name_and_source` → `_get_segment_name_and_source`

### ⚠️ Potential issues
1. **Django ASGI test** — The test sets `comm.scope["root_path"] = "/root"` but Django defaults to `path_includes_root_path=True`. Does Django ASGI actually include root_path in path? If not, this test might pass but produce incorrect URLs in production. Worth verifying Django's ASGI behavior.
2. **No test for `path_includes_root_path=False`** — The Starlette/FastAPI tests only exercise the `True` path (Starlette >= 0.33). The old behavior (`False`) has no dedicated test.
3. **Edge case: empty path** — When `path_includes_root_path=True` and `scope["path"]` is empty, the URL would be just scheme+host with no path. Unlikely but worth noting.

### Verdict
Solid PR. The core logic is correct and well-structured. The main risk is the Django default assumption — worth double-checking before merging.

**Review by ericapisani — APPROVED** [review]

File: sentry_sdk/integrations/asgi.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexander-alderman-webb) [reply sentry_sdk/integrations/asgi.py]: yes but it's ASGI spec that it's included
